efb_telegram_master/cache.py

```diff
@@ -0,0 +1,55 @@
+# coding: utf-8
+
+import weakref, collections, time
+
+class LocalCache():
+    notFound = None
+
+    # list dict等不支持弱引用，但其子类支持，故这里包装了下
+    class Dict(dict):
+        def __del__(self):
+            pass
+
+    def __init__(self, maxlen=20):
+        self.weak = weakref.WeakValueDictionary()
+        self.strong = collections.deque(maxlen=maxlen)
+
+    @staticmethod
+    def nowTime():
+        return int(time.time())
+
+    def get(self, key):
+        _ = self.weak.get(key, self.notFound)
+        if _ is not self.notFound:
+            expire = _[r'expire']
```

Comment:
`r` prefix of strings are used to disable escaping sequences. It is not useful here, same to the next line.
```suggestion
            expire = _['expire']
```